tests/test_mcmc_helper.py

```diff
@@ -43,27 +46,109 @@ def test_mcmc_helper_parameters():
 
 
 def test_mcmc_helper_target_function():
+    """ Set up priors on the model parameters such that we can
+    readily compute their expected values. """
     data = build_data()
     model = build_model(data)
 
+    prior_width = 200
+
     hmc_helper = gpflow.optimizers.SamplingHelper(
         model.trainable_parameters, model.log_marginal_likelihood
     )
+    target_log_prob_fn = hmc_helper.target_log_prob_fn
+
+    # Priors which are set on the constrained space
+    expected_log_prior = 0.
+    for param in model.trainable_parameters:
+        if param.value() < 1e-3:
+            param.assign(1.0)
+
+        param.transform = Exp()
+
+        low_value = -100
+        high_value = low_value + prior_width
+
+        param.prior = Uniform(low=np.float64(low_value), high=np.float64(high_value))
```

Comment:
```suggestion
        param.prior = Uniform(low=np.float64(low_value), high=np.float64(high_value))
        param.prior_on = gpflow.base.PriorOn.CONSTRAINED
```
just in case the default ever changes, we want both cases to be tested explicitly...